File: playground/sangjun121/java-springboot-order-perf-optimizer/src/main/java/com/order/perf/domain/Product.java
package com.order.perf.domain;

import com.order.perf.common.BaseEntity;
import lombok.Getter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Getter
@Table(name = "PRODUCTS")
@Entity
public class Product extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "product_id", nullable = false)
    private Long id;

    @Column(name = "product_name", nullable = false)
    private String productName; // 상품 이름

    @Column(name = "description", nullable = false)
    private String description; // 상품 설명

    @Column(name = "price", nullable = false)
    private int price; // 상품 가격

    @Column(name = "bundle_name", nullable = false)
    private String bundleName;   // 상품 묶음 이름 (예: 크리스마스 선물 세트)

    @Column(name = "bundle_quantity", nullable = false)
    private int bundleQuantity;  // 해당 묶음에 포함된 상품의 수량
    protected Product() {
    }

    public Product(
            final String productName, final String description,
            final int price,
            final String bundleName, final int bundleQuantity) {
        this.productName = productName;
        this.description = description;
        this.price = price;
        this.bundleName = bundleName;
        this.bundleQuantity = bundleQuantity;
    }
}

File: playground/ruthetum/app/e-commerce/order/src/main/kotlin/team/sipe/spurt/ecommerce/order/domain/Product.kt
package team.sipe.spurt.ecommerce.order.domain

import jakarta.persistence.*
import org.hibernate.annotations.Comment

@Table(name = "products")
@Entity
class Product(
    @Comment("상품 이름")
    @Column(name = "product_name")
    val productName: String,

    @Comment("상품 설명")
    @Column(name = "description")
    val description: String,

    @Comment("상품 가격")
    @Column(name = "price")
    val price: Int,

    @Comment("상품 묶음 이름")
    @Column(name = "bundle_name")
    val bundleName: String,

    @Comment("상품 묶음 수량")
    @Column(name = "bundle_quantity")
    val bundleQuantity: Int,

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
) : team.sipe.spurt.ecommerce.order.common.BaseEntity()